include/nbl/asset/utils/CSPIRVIntrospector.h

```diff
@@ -526,6 +527,30 @@ class NBL_API2 CSPIRVIntrospector : public core::Uncopyable
 
 					return range;
 				}
+				inline SDescriptorArrayInfo addDescriptorCount(const uint32_t size, const bool sizeIsLiteral)
+				{
+					const auto descCount = alloc<SDescriptorArrayInfo>(1u);
+					auto descriptorArraySize = descCount(m_memPool.data());
```

Comment:
don't allocate anything, no point allocating fixed-size member, the alloc pool stuff is only for variable lenght things like strings, and arrays of unknown size